pycape/cape.py

```diff
@@ -129,6 +130,33 @@ async def connect(
         function_ref = _convert_to_function_ref(function_ref)
         await self._request_connection(function_ref, pcrs)
 
+    @_synchronizer
+    async def deploy(self, deploy_path: Union[str, os.PathLike]) -> fref.FunctionRef:
+        """Deploy a directory or a zip file containing a Cape function declared in
+        an app.py script.
+
+        This method calls `cape deploy` from the Cape CLI to deploy a Cape function
+        then returns a `~.function_ref.FunctionRef` representingthe the deployed
+        function.  Note that the ``deploy_path`` has to point to a directory or a
+        zip file containing a Cape function declared in an app.py file and the size
+        of its content  is currently limited to 1GB.
+
+        Args:
+            deploy_path: A path pointing to a directory or a zip file containing
+            a Cape function declared in an app.py script.
+
+        Returns:
+            A :class:`~.function_ref.FunctionRef` representing the deployed Cape
+            function.
+
+        Raises:
+            RuntimeError: if the websocket response or the enclave attestation doc is
+                malformed, or if the function path is not pointing to a directory
+                or a zip file or if folder size exceeds 1GB, or if the Cape CLI cannot
+                be found on the device.
+        """
+        return await self._deploy(deploy_path)
```

Comment:
i think this indirection into a separate `_deploy` method isn't necessary, you could just take the code from `_deploy` and move it here